# Sum of Permutations of a Set

Ravi Dayabhai

Consider the following question (from [this YouTube video](https://www.youtube.com/watch?v=wamTGA2R68s) from MindYourDecisions):

Let $S$ be the set $\lbrace 1, 2, 3, 4, 5 \rbrace$. Consider the five digit base-$10$ numbers that can be formed by elements from this set such that each digit is unique. What is the sum of all such five digit numbers?

## Numerical Answer

Given that $5!$ is a relatively small number, we can brute force a solution:

In [1]:
from itertools import permutations
from functools import partial
from math import factorial

In [2]:
S = "12345"
P = permutations(S)
total = sum(map(lambda p: int(''.join(partial(map, str)(p))), P))

total

3999960

## Analytical Answer

We can leverage symmetry. Let $d_{i}$ be the digit in the $i$th position where $i \in \lbrace 0, 1, 2, 3, 4 \rbrace$ and $i = 0$ corresponding to the most significant digit. The sum of any permutation is given by:

$$
10000d_{0} + 1000d_{1} + 100d_{2} + 10d_{3} + 1d_{4}
$$

If we let $D_{i}$ be the sum of $d_{i}$ over the set of permutations, the total sum we desire can be expressed as 

$$
10000D_{0} + 1000D_{1} + 100D_{2} + 10D_{3} + 1D_{4}
$$

Each element in $S$ will occur in $i$th place for $4!$ permutations. Thus, $D_{i} = (1 + 2 + 3 + 4 + 5)(4!) = 15(24) = 360$ for all $i$, giving:

$$
10000(360) + 1000(360) + 100(360) + 10(360) + 1(360) = 3999960
$$

In [3]:
D = 15 * factorial(4)
total = sum([D * 10**k for k in range(5)])

total

3999960